feat(helm-chart): Released chart for v0.14.0

appkins opened the pull request:
<!--
    Please read https://github.com/kubernetes-sigs/external-dns#contributing before submitting
    your pull request. Please fill in each section below to help us better prioritize your pull request. Thanks!
-->

**Description**

This PR release the helm chart for `v0.14.0`.

Fixes #4072

**Checklist**

- [ ] Unit tests updated
- [ ] End user documentation updated

stevehipwell reviewed (changes requested):
Thanks for the PR @appkins. We're currently waiting on some other work before we release the chart. The PR to do so needs to have the CHANGELOG correctly updated and the changes from this added to the _Chart.yaml_ annotations (replacing the existing ones). This chart release would also be the `v1.14.0` release.

If you'd like to update your PR to reflect these changes I'll DM you when the other work has been merged so you can rebase & update the chart annotations and this PR can be the release PR.

appkins commented:
> Thanks for the PR @appkins. We're currently waiting on some other work before we release the chart. The PR to do so needs to have the CHANGELOG correctly updated and the changes from this added to the _Chart.yaml_ annotations (replacing the existing ones). This chart release would also be the `v1.14.0` release.
> 
> If you'd like to update your PR to reflect these changes I'll DM you when the other work has been merged so you can rebase & update the chart annotations and this PR can be the release PR.

I changed the release version in the PR to `v1.14.0`. Keep me posted, I'm happy to include the other changes once they're merged.

ruben-boada-klarrio commented:
Hi there! Is any plan to merge it soon? We would like to use this new version as it contains some fixes that we're waiting for. Thanks!! :wink:

stevehipwell commented:
@ruben-boada-klarrio this PR is waiting for #4032. In the meanwhile you can use the `v0.14.0` image in the currently released chart.

stevehipwell reviewed (changes requested):
@appkins could you add all of the CHANGELOG entries to the chart annotations.

appkins commented:
> @appkins could you add all of the CHANGELOG entries to the chart annotations.

I added the CHANGELOG entries to the Chart.yaml.

Raffo commented:
/ok-to-test

Raffo commented:
The linter is asking for some fixes, can you get to them @appkins ?

stevehipwell reviewed (changes requested):
@appkins you're missing the following CHANGELOG entry and you will need to run `helm-docs` locally to get linting to pass.

```yaml
    - kind: changed
      description: "Avoid unnecessary pod restart on each helm chart version. ([#4103](https://github.com/kubernetes-sigs/external-dns/pull/4103)) [@jkroepke](https://github.com/jkroepke)"

appkins commented:
> @appkins you're missing the following CHANGELOG entry and you will need to run `helm-docs` locally to get linting to pass.
> 
> ```yaml
>     - kind: changed
>       description: "Avoid unnecessary pod restart on each helm chart version. ([#4103](https://github.com/kubernetes-sigs/external-dns/pull/4103)) [@jkroepke](https://github.com/jkroepke)"
> ```

I added this line to the changelog.



> The linter is asking for some fixes, can you get to them @appkins ?

Looks like it needed a local run of `helm-docs`. I updated the documentation and pushed the changes.

stevehipwell commented:
/lgtm
/approve
